test/junit/scala/tools/nsc/reporters/ConsoleReporterTest.scala

```diff
@@ -0,0 +1,163 @@
+    }
+    writerOut.reset
+  }
+
+  
+  @Test
+  def printMessageTest(): Unit = {
+    testHelper(msg = "Hello World!")(_ => reporter.printMessage("Hello World!"))
+    testHelper(msg = "Testing with NoPosition")(_ => reporter.printMessage(NoPosition, "Testing with NoPosition"))
+    testHelper(posWithSource, "Testing with Defined Position")(_ => reporter.printMessage(posWithSource, "Testing with Defined Position"))
+  }
+
+
+  @Test
+  def echoTest(): Unit = {
+    reporter.echo("Hello World!")
+    assertEquals("Hello World!", echoWriterOut.toString.lines.next)
+
+    /** Check with constructor which has the same writer and echoWriter */
+    val reporter2 = createConsoleReporter("r", writerOut)
+    testHelper(msg = "Hello World!")(_ => reporter2.echo("Hello World!"))
+  }
+
+
+  @Test
+  def printTest(): Unit = {
+    testHelper(msg = "test")(_ => reporter.print(NoPosition, "test", reporter.INFO))
+    testHelper(msg = "test", severity = "warning: ")(_ => reporter.print(NoPosition, "test", reporter.WARNING))
+    testHelper(msg = "test", severity = "error: ")(_ => reporter.print(NoPosition, "test", reporter.ERROR))
+    testHelper(posWithSource, msg = "test")(_ => reporter.print(posWithSource, "test", reporter.INFO))
+    testHelper(posWithSource, msg = "test", severity = "warning: ")(_ => reporter.print(posWithSource, "test", reporter.WARNING))
+    testHelper(posWithSource, msg = "test", severity = "error: ")(_ => reporter.print(posWithSource, "test", reporter.ERROR))
+  }
+
+
+  @Test
+  def printColumnMarkerTest(): Unit = {
+    testHelper(msg = "")(_ => reporter.printColumnMarker(NoPosition))
+
+    reporter.printColumnMarker(posWithSource)
+    assertEquals("    ^", writerOut.toString.lines.next)
+    writerOut.reset
+  }
+
+
+  @Test
+  def displayTest(): Unit = {
+    testHelper(msg = "Testing display")(_ => reporter.display(NoPosition, "Testing display", reporter.INFO))
+    testHelper(msg = "Testing display", severity = "warning: ")(_ => reporter.display(NoPosition, "Testing display", reporter.WARNING))
+    testHelper(msg = "Testing display", severity = "error: ")(_ => reporter.display(NoPosition, "Testing display", reporter.ERROR))
+    testHelper(posWithSource, msg = "Testing display")(_ => reporter.display(posWithSource, "Testing display", reporter.INFO))
+    testHelper(posWithSource, msg = "Testing display", severity = "warning: ")(_ => reporter.display(posWithSource, "Testing display", reporter.WARNING))
+    testHelper(posWithSource, msg = "Testing display", severity = "error: ")(_ => reporter.display(posWithSource, "Testing display", reporter.ERROR))
+
+    reporter.ERROR.count += 1
+    testHelper(posWithSource, msg = "Testing display for maxerrs to pass", severity = "error: ")(_ => reporter.display(posWithSource, "Testing display for maxerrs to pass", reporter.ERROR))
+    reporter.ERROR.count += 1
+    testHelper(msg = "")(_ => reporter.display(posWithSource, "Testing display for maxerrs to fail", reporter.ERROR))
```

Comment:
Since you're testing `display` (and not `reporter.error`), I'd set `maxerrs = 1` and `ERROR.count = 0` explicitly in the test so we can see the conditions of the test. I don't know what the current count is, does it depend on order tests are run? etc.